### PR TITLE
More serial rename fixes

### PR DESCRIPTION
**Pull request by wom-bat**
Plus get rid of need for sddf_printf by defining our own assert() macro

### CODE REVIEW

File: drivers/serial/arm/uart.c [hunk 1]
```diff
@@ -17,6 +16,18 @@
 #define LOG_DRIVER(...) do{ microkit_dbg_puts(microkit_name); microkit_dbg_puts("|INFO: "); microkit_dbg_puts(__VA_ARGS__); }while(0)
 #define LOG_DRIVER_ERR(...) do{ microkit_dbg_puts(microkit_name); microkit_dbg_puts("|ERROR: "); microkit_dbg_puts(__VA_ARGS__); }while(0)
 
+static void assert_fail(const char *s, int line)
```
Comment (Ivan-Velickovic) [inline drivers/serial/arm/uart.c:19]: Don't think we should be doing this in a driver.

**Review by Ivan-Velickovic — CHANGES_REQUESTED** [review]

**Comment by Ivan-Velickovic** [comment]
I think these fixes are already in https://github.com/au-ts/sDDF/pull/65.

File: drivers/serial/arm/uart.c (hunk 1 above)
Comment (wom-bat) [reply drivers/serial/arm/uart.c:19]: Looks like this has been fixed elsewhere.

**Comment by wom-bat** [comment]
This is covered in a different PR

**Pull request closed without merging**